Frends.AzureBlobStorage.UploadBlob/Frends.AzureBlobStorage.UploadBlob/Definitions/Destination.cs

```diff
@@ -1,91 +1,134 @@
+    /// </summary>
+    /// <example>true</example>
+    [DefaultValue(true)]
+    public bool CreateContainerIfItDoesNotExist { get; set; }
+
+    /// <summary>
+    /// Azure blob type to upload: Append, Block or Page.
+    /// </summary>
+    /// <example>Block</example>
+    [DefaultValue(AzureBlobType.Block)]
+    public AzureBlobType BlobType { get; set; }
+
+    /// <summary>
+    /// Specifies the maximum size for the Page blob, up to 8 TB. The size must be aligned to a 512-byte boundary (512, 1024, 1536..). Calculating minimum value from file's size if given value is less than 512.
+    /// </summary>
+    /// <example>1024</example>
+    [UIHint(nameof(BlobType), "", AzureBlobType.Page)]
+    public long PageMaxSize { get; set; }
+
+    /// <summary>
+    /// Specifies the starting offset for the content to be written as a Page. Value range from 0 to 'page max-file size'. If set to -1, 'page max-file size' will be calculated from 'PageMaxSize'-file's size.  
+    /// </summary>
+    /// <example>0</example>
+    [UIHint(nameof(BlobType), "", AzureBlobType.Page)]
+    public long PageOffset { get; set; }
+
+    /// <summary>
+    /// Source file can be renamed to this name in Azure Blob Storage.
+    /// </summary>
+    /// <example>Renamed.txt</example>
+    [DisplayFormat(DataFormatString = "Text")]
+    public string RenameTo { get; set; }
+
+    /// <summary>
+    /// Set desired content-type. If empty, task tries to guess from mime-type.
+    /// </summary>
+    /// <example>text/xml</example>
+    public string ContentType { get; set; }
+
+    /// <summary>
+    /// Set desired content-encoding. Defaults to UTF8 BOM.
+    /// </summary>
+    /// <example>utf8</example>
+    public string FileEncoding { get; set; }
+
+    /// <summary>
+    /// Should upload operation overwrite existing blob with same name?
+    /// </summary>
+    /// <example>false</example>
+    [DefaultValue(false)]
+    public bool Overwrite { get; set; }
+
+    /// <summary>
+    /// Append blob with 'Source File'. Block and Page blob will be downloaded into 'Download Folder' and uploaded back into same container after local append process is completed. No downloading needed for Append Blob. Overwrite must be true when targeting Block or Page blob.
+    /// </summary>
+    /// <example>false</example>
+    [DefaultValue(false)]
+    public bool Append { get; set; }
+
+    /// <summary>
+    /// Blob's name. 'Source File' will be appended into this blob.
+    /// </summary>
+    /// <example>TestFile.txt</example>
+    [UIHint(nameof(Append), "", true)]
+    public string BlobName { get; set; }
+
+    /// <summary>
+    /// Directory where blob will be downloaded for appending process. Only Block and Page blobs will be downloaded.
+    /// </summary>
+    /// <example>c:/temp/downloads</example>
+    [UIHint(nameof(Append), "", true)]
+    public string DownloadFolder { get; set; }

```

Comment:
User should not be concerned about the appending process, but everything should happen behind the scenes. You should either download the file to temp and delete it after append is done or read the blob content straight to memory: https://stackoverflow.com/questions/62539924/download-blob-file-into-memory-stream-from-azure-using-c-sharp

Reply:
Going to use downloading into temp file.